mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -31,7 +31,8 @@ import org.apache.spark.sql.DataFrame
  * @param predictionAndObservations an RDD of (prediction, observation) pairs.
  */
 @Experimental
-class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)]) extends Logging {
+class RegressionMetrics(@transient predictionAndObservations: RDD[(Double, Double)])
+  extends Logging with Serializable {
```

Comment:
I agree that `RegressionMetrics` will probably not need to be serialized and sent to workers. However, `LinearRegressionModel` needs to be serializable => `LinearRegressionSummary` and its fields need to as well.